src/main/java/com/example/ajaxonboardingproject/nats/NatsMovieSessionUpdateController.kt

```diff
@@ -24,12 +23,15 @@ class NatsMovieSessionUpdateController(
         MovieSessionUpdateRequest.parser()
 
     override fun generateReplyForNatsRequest(
-        request: MovieSessionUpdateRequest
+        request: Mono<MovieSessionUpdateRequest>
     ): Mono<MovieSessionResponse> {
-        val movieSession: MovieSession = converter.protoToMovieSession(request.movieSession).apply {
-            id = request.id
-        }
-        return service.update(movieSession)
+        return request
+            .map {
+                request -> converter.protoToMovieSession(request.movieSession).apply {
+                    id = request.id
+                }
+            }
```

Comment:
```
override fun generateReplyForNatsRequest(request: Mono<MovieSessionUpdateRequest>): Mono<MovieSessionResponse> =
    request.map { converter.protoToMovieSession(it.movieSession).apply { id = it.id } }
           .flatMap(service::update)
           .map(converter::movieSessionToProtoResponse)
```

Maybe using the implicit lambda parameter **it** and method references. 
Also **request** as a lambda parameter is a bit ambiguous because function also has a parameter named **request**.

Reply:
I agree that your approach looks more neat, but I'm ok with the two approaches.

Reply:
I strongly agree with the formatting that Arsen suggested
```
request.map { converter.protoToMovieSession(it.movieSession).apply { id = it.id } }
```

or 

```
request.map { request ->
    converter.protoToMovieSession(it.movieSession).apply { id = request.id } 
}
```

up to you

Reply:
Done! ✅